### PR TITLE
Update `clavrx` reader file patterns and flag_meanings handling

### LINKED ISSUE #2204: CLAVRx Reader for AHI data misses CLAVRx files with naming convention that retains full AHI original name.
**Describe the bug**
The CLAVRx reader has based the input AHI filename on a shortened version of that filename which typically is not generated by the CLAVRx processing routine, but is a branch that usually does not survive updates.  Therefore, files routinely produced by the CLAVRx processing routine are not found with find_files_and_readers.
**To Reproduce**

Attempting to load a filename of clavrx_H08_20220920_1230_B01_FLDK_DK_R10_S0110.DAT.level2.nc in a directory which also contains files with the filename convention of clavrx_H08_202209181230_B01_FLDK_DK_R.nc in a local path represented by 
<path to local AHI data>
```python
# Your code here

from satpy import find_files_and_readers
f = find_files_and_readers(base_dir=<path to local AHI data>

```

**Expected behavior**
Ether filename convention should be discovered by the CLAVRx reader.
**Actual results**
Only the filename which was actually added to the setup file for the CLAVRx reader is found: _clavrx_H08_202209181230_B01_FLDK_DK_R.nc_

**Environment Info:**
- All Environments (oversight in reader)
 - Readers and writers dependencies (when relevant): clavrx reader dependency

### PR DESCRIPTION
**Pull request by joleenf**
Update filenames in clavrx.yaml to match two names that were recently discovered as default output from the CLAVRx processing routine.  
Update flag_meanings in clavrx.py so that they are returned as a list for easier pairing with the flag_values.

Edit By @djhoese: This also adds GOES-18 handling and adds explicit AWIPS handling of the `insolation_dcomp` product. 

 - [ ] Closes #2204
 - [ ] No new tests added

### CODE REVIEW

**Review by djhoese — COMMENTED** [review]
The new file pattern by itself probably doesn't require tests, but the new SCALED handling probably should have tests, right?

File: satpy/etc/readers/clavrx.yaml [hunk 1]
```diff
@@ -1,10 +1,6 @@
 reader:
-  name: clavrx
-  short_name: CLAVR-X
-  long_name: The Clouds from AVHRR Extended (CLAVR-x)
   description: CLAVR-X Reader
-  status: Nominal
-  supports_fsspec: false
+  name: clavrx
```
Comment (mraspaud) [inline satpy/etc/readers/clavrx.yaml:3]: Is the removal of the metadata here intentional?

File: satpy/readers/caliop_l2_cloud.py.old [hunk 2]
```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
```
Comment (mraspaud) [inline satpy/readers/caliop_l2_cloud.py.old:1]: what is the meaning of this .old file?

File: satpy/readers/clavrx.py [hunk 3]
```diff
@@ -70,6 +70,7 @@
 
 def _get_sensor(sensor: str) -> str:
     """Get the sensor."""
+    LOG.debug('SENSORS: {}'.format(sensor))
```
Comment (mraspaud) [inline satpy/readers/clavrx.py:73]: an f-string would be more concise, eg
```suggestion
    LOG.debug(f'SENSORS: {sensor}')
```

File: satpy/readers/clavrx.py [hunk 4]
```diff
@@ -92,29 +93,29 @@ def _get_rows_per_scan(sensor: str) -> Optional[int]:
...
+        """Remove attributes that described data before scaling."""
+        old_attrs = ['unscaled_missing', 'SCALED_MIN', 'SCALED_MAX',
+                     'SCALED_MISSING']
+
+        for attr_key in old_attrs:
+            attrs.pop(attr_key, None)
+        return attrs
+
+    @staticmethod
+    def _scale_data(data_arr, scale_factor: float, add_offset: float) -> xr.DataArray:
```
Comment (mraspaud) [inline satpy/readers/clavrx.py:110]: Is the removal of the type hint on `data_arr` intentional?

File: satpy/readers/clavrx.py [hunk 5]
```diff
@@ -162,7 +163,7 @@ def _area_extent(x, y, h):
         return area_extent, ncols, nlines
 
     @staticmethod
-    def _read_pug_fixed_grid(projection, distance_multiplier=1.0) -> dict:
+    def _read_pug_fixed_grid(projection: xr.DataArray, distance_multiplier=1.0) -> dict:
```
Comment (mraspaud) [inline satpy/readers/clavrx.py:166]: isn't `projection` a CRS object?

File: satpy/readers/clavrx.py [hunk 6]
```diff
@@ -261,6 +261,8 @@ def get_metadata(sensor, platform, attrs: dict, ds_info: dict) -> dict:
         if not i.get('SCALED', 1) and not flag_meanings:
             i['flag_meanings'] = '<flag_meanings_unknown>'
             i.setdefault('flag_values', [None])
+        elif not i.get('SCALED', 1) and isinstance(flag_meanings, str):
```
Comment (mraspaud) [inline satpy/readers/clavrx.py:264]: `i` is not a good name for this variable, please use something more relevant.

File: satpy/tests/reader_tests/test_clavrx.py [hunk 7]
```diff
@@ -215,18 +226,23 @@ def test_load_all(self):
...
             self.assertEqual(v.attrs['platform_name'], 'npp')
             self.assertEqual(v.attrs['sensor'], 'viirs')
             self.assertIsInstance(v.attrs['area'], SwathDefinition)
             self.assertEqual(v.attrs['area'].lons.attrs['rows_per_scan'], 16)
             self.assertEqual(v.attrs['area'].lats.attrs['rows_per_scan'], 16)
-        self.assertIsNotNone(datasets['variable3'].attrs.get('flag_meanings'))
+        flag_meanings = datasets["variable3"].attrs.get("flag_meanings")
+        if isinstance(flag_meanings, str):
+            self.assertFalse(" " in flag_meanings)
+            self.assertTrue(flag_meanings == "<flag_meanings_unkown>")
```
Comment (mraspaud) [inline satpy/tests/reader_tests/test_clavrx.py:243]: is this code tested?

File: satpy/tests/reader_tests/test_clavrx.py [hunk 8]
```diff
@@ -104,6 +104,16 @@ def get_test_content(self, filename, filename_info, filetype_info):
...
+        # category2 Flag meanings not available (should default to flag meanings unknown)
+        file_content['variable4'] = xr.DataArray(
+            da.from_array(DEFAULT_FILE_DATA, chunks=4096).astype(np.byte),
+            attrs={
+                'SCALED': 0,
+                '_FillValue': -128,
+                'units': 'none',
+            })
+
+        file_content['variable4/shape'] = DEFAULT_FILE_SHAPE
```
Comment (mraspaud) [inline satpy/tests/reader_tests/test_clavrx.py:116]: Codescene flags code duplication here, is this correct?

**Review by mraspaud — CHANGES_REQUESTED** [review]

File: satpy/etc/readers/clavrx.yaml (hunk 1 above)
Comment (joleenf) [reply satpy/etc/readers/clavrx.yaml:3]: Should I pull in upstream/main before the next commit?

File: satpy/readers/clavrx.py (hunk 5 above)
Comment (joleenf) [reply satpy/readers/clavrx.py:166]: No, not in this case.  This method is reading the x,y fixed grid and attributes information from a netCDF files.  So, this is actually a netCDF4 variable by the way it is being read.

File: satpy/tests/reader_tests/test_clavrx.py (hunk 7 above)
Comment (joleenf) [reply satpy/tests/reader_tests/test_clavrx.py:243]: I thought so, yes, but clearly, no.

File: satpy/readers/clavrx.py (hunk 4 above)
Comment (djhoese) [reply satpy/readers/clavrx.py]: If the valid_range isn't a DataArray then I guess `Union[xr.DataArray, list, tuple]`. Or you could always force `valid_range` to a tuple when calling this method and then do `Union[xr.DataArray, tuple[float, float]]` (or remove the sub-types on the tuple).

File: satpy/etc/readers/clavrx.yaml (hunk 1 above)
Comment (djhoese) [reply satpy/etc/readers/clavrx.yaml:3]: I'm not sure what pulling in upstream matters here. I guess maybe your base `main` branch was old? You should maybe merge upstream pytroll `main` into your PR. That should also fix the experimental CI environment that is failing.

**Comment by djhoese** [comment]
@joleenf Sorry we got lost on this again. What's the current state of this PR? Is there still work to be done to address Martin's concerns? Or is it ready for re-review?

File: satpy/readers/clavrx.py (hunk 5 above)
Comment (mraspaud) [reply satpy/readers/clavrx.py]: then I suggest this parameter to be rename, eg to `projection_coordinates`

File: satpy/etc/readers/clavrx.yaml (hunk 1 above)
Comment (joleenf) [reply satpy/etc/readers/clavrx.yaml:3]: No.  That was not intentional.

**Comment by djhoese** [comment]
Looks like you'll need to merge with `main` to get some of the `np.bool` fixes so that the docs build with the newest numpy versions.

File: satpy/readers/clavrx.py [hunk 9]
```diff
@@ -70,10 +71,11 @@
 
 def _get_sensor(sensor: str) -> str:
     """Get the sensor."""
+    LOG.debug(f'SENSORS: {sensor}')
```
Comment (djhoese) [inline satpy/readers/clavrx.py:74]: We probably don't need this log message in the long term do we?

File: satpy/readers/clavrx.py [hunk 10]
```diff
@@ -318,7 +316,9 @@ def get_nadir_resolution(self, sensor):
 
     def available_datasets(self, configured_datasets=None):
         """Automatically determine datasets provided by this file."""
+        LOG.debug(self.file_content.get('/attr/sensor'))
         self.sensor = _get_sensor(self.file_content.get('/attr/sensor'))
+        LOG.debug(self.sensor)
```
Comment (djhoese) [inline satpy/readers/clavrx.py:321]: Do we need these debug messages?

**Review by djhoese — APPROVED** [review]
Just some small comments about removing debug log messages, but otherwise I assume you've tested reading and AWIPS support so if that's true, let me know when you're ready to have this merged.

**Comment by djhoese** [comment]
This looks good to me. Any more expected changes @joleenf or can I merge this?

**Comment by djhoese** [comment]
Nah this PR already includes more than can be contained in the title. Merging now. Thanks.

**Pull request merged**